### DISCUSSION TITLE
Make properties of `QueuedQuery` and `Batch` public

### DISCUSSION THREAD

**Discussion by pashagolub** [jackc/pgx#1878] [Ideas]
I don't know what is the ratio behind having those properties being private.
My proposal is to make those public to allow programmer, at least, to read those:

```go
type QueuedQuery struct {
	Query     string
	Arguments []any
	fn        batchItemFunc
	sd        *pgconn.StatementDescription
}

type Batch struct {
	QueuedQueries []*QueuedQuery
}
```
Another non-breaking option would be to introduce a public interface with appropriate methods, just like you did for `BatchResults`.

My use case is [implementing a support for batches](https://github.com/pashagolub/pgxmock/pull/179) in [pgxmock](https://github.com/pashagolub/pgxmock). At the moment we need to use dirty hacks to get the content of the queued queries in batch, e.g.

```go
...
			qqValue := v.FieldByName("queuedQueries").Index(i)
			qq := reflect.NewAt(qqValue.Type(), unsafe.Pointer(qqValue.UnsafeAddr())).Elem().Interface().(*pgx.QueuedQuery)

			query := reflect.ValueOf(*qq).FieldByName("query").String()
			if err := c.queryMatcher.Match(q.expectSQL, query); err != nil {
				return err
			}

			argsValue := reflect.ValueOf(*qq).FieldByName("arguments")
			args := reflectBatchElementArguments(argsValue)
...
```

**Comment by jackc** [comment]
> I don't know what is the ratio behind having those properties being private.

There was no reason specific for these particular properties. Rather it was the general principle of minimizing the API surface area and avoiding being locked into any particular implementation.

But I suppose the implementation or interface is unlikely to change at this point -- so I have no objections to making those properties public.
